assets/scripts/dialogs/__tests__/GeotagDialog.test.js

```diff
@@ -17,10 +20,165 @@ describe('GeotagDialog', () => {
     expect(wrapper.exists()).toEqual(true)
   })
 
-  it('does not allow a location to be confirmed when geocoded data does not have street data')
-  it('allows a location to be confirmed when the current signed-in user is the street owner')
-  it('allows a location to be confirmed when the current anonymous user started this street')
-  it('does not allow a location to be confirmed when the current signed-in user is not the street owner')
-  it('does not allow a location to be confirmed when the current anonymous user is not the street owner and there is already an existing location attached')
-  it('allows a location to be confirmed when the current anonymous user is not the street owner but there is no existing location attached')
+  it('does not allow a location to be confirmed when geocoded data does not have street data', () => {
+    const testMarker = { lat: 0, lng: 0 }
+    const testAddressInfo = {
+      street: null
+    }
+    const wrapper = shallow(
+      <GeotagDialog.WrappedComponent
+        street={{}}
+        addressInformation={testAddressInfo}
+        markerLocation={testMarker}
+      />
+    )
+    expect(wrapper.find('button .confirm-button')).toHaveLength(0)
+  })
+  it('allows a location to be confirmed when the current signed-in user is the street owner', () => {
+    const testMarker = { lat: 0, lng: 0 }
+    const testAddressInfo = {
+      street: 'test street',
+      id: 'test id'
+    }
+    const testStreet = {
+      creatorId: 'test creator',
+      location: {
+        label: 'test location',
+        wofId: 'test id'
+      }
+    }
+    const wrapper = shallow(
+      <GeotagDialog.WrappedComponent
+        street={testStreet}
+        addressInformation={testAddressInfo}
+        markerLocation={testMarker}
+      />
+    )
+    getRemixOnFirstEdit.mockReturnValueOnce(false)
+    wrapper.setProps({
+      markerLocation: { lat: 10, lng: 10 },
+      addressInformation: {
+        street: 'test street 2',
+        id: 'test id 2'
+      }
+    })
+    expect(wrapper.find('button .confirm-button')).toHaveLength(1)
+  })
+  it('allows a location to be confirmed when the current anonymous user started this street', () => {
+    const testMarker = { lat: 0, lng: 0 }
+    const testAddressInfo = {
+      street: 'test street',
+      id: 'test id'
+    }
+    const testStreet = {
+      creatorId: null,
+      location: {
+        label: 'test location',
+        wofId: 'test id'
+      }
+    }
+    const wrapper = shallow(
+      <GeotagDialog.WrappedComponent
+        street={testStreet}
+        addressInformation={testAddressInfo}
+        markerLocation={testMarker}
+      />
+    )
```

Comment:
One thing you could probably do to reduce the amount of repeated set-up boilerplate is to write a function that returns the component to shallow render. e.g.

```js
function getTestComponent () {
   // set up the test data
   // return <GeotagDialog />
}

// later

it('does something', () => {
  const wrapper = shallow(getTestcomponent())
  // ...proceed with the rest of the test
})

Reply:
ahh thanks! I did not know I could do this.